src/libsync/connectionvalidator.cpp

```diff
@@ -247,10 +248,22 @@ void ConnectionValidator::slotAuthSuccess()
 
 void ConnectionValidator::checkServerCapabilities()
 {
+    // The main flow now needs the capabilities
     JsonApiJob *job = new JsonApiJob(_account, QLatin1String("ocs/v1.php/cloud/capabilities"), this);
     job->setTimeout(timeoutToUseMsec);
     QObject::connect(job, &JsonApiJob::jsonReceived, this, &ConnectionValidator::slotCapabilitiesRecieved);
     job->start();
+
+    // And we'll retrieve the ocs config in parallel
+    auto ocsConfigUrl = Utility::concatUrlPath(_account->url(), QLatin1String("ocs/v1.php/config"));
+    auto configJob = _account->sendRequest("GET", ocsConfigUrl);
+    configJob->setTimeout(timeoutToUseMsec);
+    // note that 'this' might be destroyed before the job finishes
+    auto account = _account;
+    QObject::connect(configJob, &SimpleNetworkJob::finishedSignal, _account.data(),
+        [=](QNetworkReply *reply) {
+            ocsConfigReceived(reply, account);
```

Comment:
also be carefull that this might actually be deleted here.

Reply:
I know, that's exactly why I'm capturing `account` by value and have `ocsConfigReceived` as a static method; will improve comments stating that.